java/src/com/ibm/streamsx/topology/internal/context/remote/ZippedToolkitRemoteContext.java

```diff
@@ -128,22 +131,33 @@ private static Path pack(final File appTkRoot, JsonObject submission, String tkN
         Path manifestTmp = Files.createTempFile("manifest_tk", ".txt");
         Path mainCompTmp = Files.createTempFile("main_composite", ".txt");
         Path scOptsTmp = Files.createTempFile("sc_opts", ".txt");
-        
+        Path splmmOptsTmp = Files.createTempFile("splmm_opts", ".txt");
+        Path splmm_dirTmp = Files.createTempFile("splmm_dir", ".txt");
+
         // tkManifest is the list of toolkits contained in the archive
         try (PrintWriter tkManifest = new PrintWriter(manifestTmp.toFile(), "UTF-8")) {
             if (tkName != null)
                 tkManifest.println(tkName);
             tkManifest.println(topologyToolkitName);
-            
             JsonObject configSpl = object(graph, CONFIG, "spl");
             if (configSpl != null) {
                 objectArray(configSpl, "toolkits",
                         tk -> {
                             if (tk.has("root")) {
                                 File tkRoot = new File(jstring(tk, "root"));
                                 String tkRootName = tkRoot.getName();
-                                tkManifest.println(tkRootName);
-                                toolkits.put(tkRoot.toPath(), tkRootName);
+                                // If SPLMM app, need to seperate this toolkit to run the command
+                                // 'spl-make-toolkit -i <SPLMM_APP_DIR> `cat splmm_opts.txt`'
+                                // Bc SPLMM app requires arguments, so can't group with other toolkits
+                                if (deploy(submission).has(ContextProperties._SPLMM_OPTIONS)) {
```

Comment:
Won't this miss toolkits if SPLMM is set? E.g. with toolkits A,B,C then A will be the SPLMM toolkits and the other two will be incorrectly ignored?

Reply:
I separate the main composite toolkit from the rest of the toolkits if it is an SPLMM app requiring arguments bc if I don't separate it, the makefile will error on this line '${STREAMS_INSTALL}/bin/spl-make-toolkit -i $${toolkit}; \'. 

ContextProperties._SPLMM_OPTIONS is only set in streamsx-sc, where if the main composite is determined to be an SPLMM app, that first toolkit (the splmm app) will be filtered out, while the rest of the toolkits will be processed by the else case as normal. Is this the correct behavior or should I be doing processing it some other way?

Reply:
How will the rest of the toolkits in that case be processed by the else?

 If the code archive has three toolkits A,B,C and _SPLMM_OPTIONS is set then A will go through lines 153-156, then B,C will fail the check at line 153 and not be added to `toolkits`.

Reply:
Oh I see, sorry about that. Will change it and re-run the tests

Reply:
Fixed in latest commit

Reply:
Fixed in latest commit